### PR TITLE
[Relax] Express dynamic arguments of strided_slice as arguments

### PR DESCRIPTION
**Pull request by Lunderberg**
Prior to this commit, `relax.op.strided_slice` stored the `axes`, `begin`, `end`, and `strides` in the `CallNode::attrs`.  However, the attributes are only intended to store static values.  The indices used used for `relax.op.strided_slice` must frequently be in terms of symbolic shape variables, which should not be stored in the attributes.  While some utilities have special handling for `relax.op.strided_slice` (e.g. `tvm::relax::Bind`), many do not (e.g. `tvm::relax::WellFormed` and
`tvm::relax::FreeSymbolicVars`).  As a result, the symbolic expressions in `relax.op.strided_slice` will fail to be updated in generic utilities, and will fail to trigger safeguards when this occurs.

This commit changes the representation of `relax.op.strided_slice` to store all arguments in the `relax::CallNode::args`, rather than the `relax::CallNode::attrs`.  As mentioned in a comment from https://github.com/apache/tvm/pull/13987, which initially implemented `relax.op.strided_slice`, this was an intended refactor once `relax::PrimValue` was fully supported.

### CODE REVIEW

**Review by slyubomirsky — APPROVED** [review]
Very good, we've discussed the topic before and I'm glad to see greater expressiveness via PrimValues where it's appropriate (as opposed to requiring things to be fixed in attributes).

File: src/relax/op/tensor/index.cc [hunk 1]
```diff
@@ -102,117 +107,326 @@ TVM_REGISTER_OP("relax.take")
...
+  // Handle Python-style negative indices
   index = if_then_else(index < 0, index + extent, index);
-  return assume_inbound ? index : min(max(index, begin_range), end_range);  // NOLINT
+  // Clamp the result to valid indices
+  PrimExpr lower_bound = tvm::if_then_else(stride < 0, -1, 0);
+  PrimExpr upper_bound = tvm::if_then_else(stride < 0, extent - 1, extent);
+  index = tvm::min(tvm::max(index, lower_bound), upper_bound);
+
+  // PrimExpr bounds_offset = tvm::if_then_else(stride < 0, -1, 0);
+  // index = tvm::min(tvm::max(index, 0 + bounds_offset), extent + bounds_offset);
```
[the remark this answers is not in the document]
Comment (Lunderberg) [reply src/relax/op/tensor/index.cc:163]: Thank you, and deleted!

**Comment by Lunderberg** [comment]
@Archermmt Can you look into the failing MSC unit tests?  It seems that it is a limitation in the Relax to MSC conversion, that it doesn't correctly handle in-line leaf nodes that occur within a `relax::Tuple`.  For now, I've marked the failing unit tests with `@pytest.mark.xfail`, but it would be good to resolve the breakages either in this or a follow-up PR.

**Comment by Lunderberg** [comment]
Merged from main into PR branch to resolve conflict.

**Pull request merged**